wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -528,50 +530,86 @@ protected void onEndTask(String taskName, int workCurr, int workTotal, int perce
     @Override
     public Revision commit(CommitParams params) throws GitException {
         try {
-            String message = params.getMessage();
-            GitUser committer = getUser();
-            if (message == null) {
-                throw new GitException("Message wasn't set");
+            // Check repository state
+            if (!repository.getRepositoryState().canCommit()) {
```

Comment:
Is `getRepositoryState` easy operation? I'd save result to variable and then reuse it